<launch>

    <!--- static tf between world and odom frames -->
    <node pkg="tf" type="static_transform_publisher" name="world_to_odom" args="0 0 0 0 0 0 1 /world odom 100" />

    <!--- differential drive simulator -->
    <node pkg="diffdrive_kin_sim" type="diffdrive_kin_sim" respawn="false" name="diffdrive_kin_sim" output="screen">
        <rosparam file="$(find diffdrive_kin_sim)/config/diffdrive_kin_sim.yaml" command="load" />
        <rosparam param="/use_sim_time">true</rosparam>
    </node>

    <!--- linearizing controller -->
    <node pkg="diffdrive_kin_ctrl" type="diffdrive_kin_trajctrl" respawn="false" name="diffdrive_kin_trajctrl" output="screen">
        <rosparam file="$(find diffdrive_kin_ctrl)/config/diffdrive_kin_trajctrl.yaml" command="load" />
        <rosparam file="$(find diffdrive_kin_sim)/config/diffdrive_kin_sim.yaml" command="load" />
    </node>

    <!--- trajectory generator -->
    <node pkg="diffdrive_kin_ctrl" type="eight_traj_gen" respawn="false" name="eight_traj_gen" output="screen">
        <rosparam file="$(find diffdrive_kin_ctrl)/config/eight_traj.yaml" command="load" />
    </node>

    <!--- RVIZ -->
    <node pkg="rviz" name="odom_visualizer" type="rviz" args="-d $(find diffdrive_kin_ctrl)/rviz/odom_visualizer.rviz"/>

</launch>
